File: android/app/src/main/java/com/pan3d/skill/vo/SkillFixEffectKeyVo.java
package com.pan3d.skill.vo;

import com.pan3d.vo.DataObjTempVo;
import com.pan3d.vo.Vector3D;

public class SkillFixEffectKeyVo extends SkillKeyVo {

    public Vector3D pos  ;
    public Vector3D rotation  ;
    public boolean hasSocket  ;
    public String socket;

    public void setData(DataObjTempVo $data) {
        super.setData($data);
        this.hasSocket = $data.hasSocket;
        if (this.hasSocket) {
            this.socket = $data.socket;
        } else {
            this.pos = new Vector3D($data.pos.x, $data.pos.y, $data.pos.z);
            this.rotation = new Vector3D($data.rotation.x, $data.rotation.y, $data.rotation.z);
        }

    }
}

File: new/src/Pan3d/skill/vo/SkillFixEffectKeyVo.ts
module Pan3d{
    export class SkillFixEffectKeyVo extends SkillKeyVo {
        public pos: Vector3D;
        public rotation: Vector3D;
        public hasSocket: boolean;
        public socket: string;

        public setData($data: any): void {
            super.setData($data);

            this.hasSocket = $data.hasSocket;

            if (this.hasSocket) {
                this.socket = $data.socket;
            } else {
                this.pos = new Vector3D($data.pos.x, $data.pos.y, $data.pos.z);
                this.rotation = new Vector3D($data.rotation.x, $data.rotation.y, $data.rotation.z);
            }

        }

    }
}

